enterprise/models.py

```diff
@@ -1365,6 +1365,50 @@ def __repr__(self):
         return self.__str__()
 
 
+@python_2_unicode_compatible
+class LicensedEnterpriseCourseEnrollment(TimeStampedModel):
+    # Would the license_uuid be a primary_key on this(?)
+    license_uuid = models.UUIDField(
+        primary_key=False,
+        editable=False,
+        null=False
+    )
+
+    enterprise_course_enrollment = models.OneToOneField(
+        EnterpriseCourseEnrollment,
+        blank=False,
+        null=False,
+        related_name='licensed_with',
+        on_delete=models.deletion.CASCADE,
+        help_text=_(
+            "The course enrollment the associated license is for."
+        )
+    )
+
+    history = HistoricalRecords()
+
+    # Does there need to be a source (?)
+    source = models.ForeignKey(
+        EnterpriseEnrollmentSource,
+        blank=False,
+        null=True,
+        on_delete=models.SET_NULL
+    )
+
+    # The ticket said this would be nice to have for returning an associated license if one exists
+    # In what cases will we have a licensed enrollment but there not being an assocaited license?
+    @property
+    def associated_license(self):
+        """
+        Returns the license associated with this enterprise course enrollment if one exists
+        """
```

Comment:
Right, you actually want to add this on the `EnterpriseCourseEnrollment` model.  Because you're adding a `OneToOneField` from this model back to `EnterpriseCourseEnrollment`, `EnterpriseCourseEnrollment` will now have an attribute named whatever the `related_name` is of the `OneToOneField` (`licensed_with`, in this case).

There's an example in the docs: https://docs.djangoproject.com/en/2.2/topics/db/examples/one_to_one/ about why this property would be useful:
```
> from django.core.exceptions import ObjectDoesNotExist
>>> try:
>>>     p2.restaurant
>>> except ObjectDoesNotExist:
>>>     print("There is no restaurant here.")
There is no restaurant here.
```